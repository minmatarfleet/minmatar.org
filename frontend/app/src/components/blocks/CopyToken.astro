---
import { i18n } from '@helpers/i18n'
const { t, translatePath } = i18n(Astro.url)

interface Props {
    position?:  'left' | 'right';
}

const {
    position = 'right',
} = Astro.props

const auth_token = Astro.cookies.has('auth_token') ? (Astro.cookies.get('auth_token')?.value as string) : false

const COPIED_LABEL_TIME_MS = 2000

import Wrapper from '@components/compositions/Wrapper.astro';

import Tippy from '@components/blocks/Tippy.astro';
import Textarea from '@components/blocks/Textarea.astro';
import StylessButton from '@components/blocks/StylessButton.astro';

import ClipboardIcon from '@components/icons/buttons/ClipboardIcon.astro';

---

<Wrapper
    class:list={[ 'copy-auth-token', 'relative', position ]}
    max_width='32px'
    padding_block='0'
    padding_inline='0'
    x-data={`{
        click_class: '',
    }`}
>
    <Tippy
        class="[ flex text-[var(--highlight)] ]"
        data-tippy-placement="right"
        partial={`${translatePath('/partials/normal_tooltip')}?description=auth_token_page`}
        height='70px'
        width='340px'
        element='div'
        show_delay={0}
    >
        <StylessButton
            class="[ !pointer-events-auto ]"
            x-bind:class="click_class"
            x-on:click.stop.prevent={`copyFitToClipboard('copy-auth-token')`}
            x-on:click.throttle.750ms={`click_class = 'glow'; setTimeout(() => click_class = '', ${COPIED_LABEL_TIME_MS})`}
            x-init="$el.removeAttribute('disabled')"
            disabled
        >
            <ClipboardIcon width={12} height={12} />
        </StylessButton>
        <Textarea class="[ hidden ]" id="copy-auth-token" readonly>{auth_token}</Textarea>
    </Tippy>
    <small class="[ animate-fade-in-up ]" x-show="click_class === 'glow'" style="display: none">{t('copied')}</small>
</Wrapper>

<style lang="scss">
    .glow {
        filter: drop-shadow(0 0 3px rgba(255, 255, 255, 0.5));
    }

    .left small {
        right: 30px;
    }

    .right small {
        left: 30px;
    }

    small {
        position: absolute;
        top: 2px;
        white-space: nowrap;
        color: var(--fleet-yellow);
    }
</style>