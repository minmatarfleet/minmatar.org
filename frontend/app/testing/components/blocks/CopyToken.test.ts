import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { expect, test, vi } from "vitest";
import CopyToken from "@components/blocks/CopyToken.astro";

test("CopyToken defaults", async () => {
  const container = await AstroContainer.create();
  const result = await container.renderToString(CopyToken, {});

  expect(result).toMatchSnapshot();
});
