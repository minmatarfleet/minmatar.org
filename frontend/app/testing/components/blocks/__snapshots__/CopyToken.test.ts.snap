// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CopyToken defaults 1`] = `
"<div class="wrapper copy-auth-token relative right" x-data="{
        click_class: '',
    }" data-astro-cid-5ivck5fi="true" data-mod data-astro-cid-yv35cmqx style="--wrapper_padding_inline: 0;--wrapper_padding_inline_end: 0;--wrapper_padding_block: 0;--wrapper_max_width: 32px;">  <div data-tippy-content="
            <div hx-on--after-request=&#34;
                const popper = this.closest('[data-tippy-root]')
                const instance = popper._tippy
                instance.popperInstance.update()
                popper.classList.remove('visually-hidden')
            &#34;>
                <span
                    hx-on--before-request=&#34;this.closest('[data-tippy-root]').classList.add('visually-hidden')&#34;
                    hx-trigger=&#34;intersect&#34;
                    hx-get='/partials/normal_tooltip?description=auth_token_page'
                    hx-swap=&#34;outerHTML transition:false&#34;
                >
                    Loading
                </span>
            </div>
        " data-tippy-delay="[0, 200]" data-tippy-allowhtml="true" class="[ flex text-[var(--highlight)] ]" data-tippy-placement="right" data-astro-cid-5ivck5fi="true"> <button class="[ !pointer-events-auto ]" x-bind:class="click_class" x-on:click.stop.prevent="copyFitToClipboard('copy-auth-token')" x-on:click.throttle.750ms="click_class = 'glow'; setTimeout(() => click_class = '', 2000)" x-init="$el.removeAttribute('disabled')" disabled data-astro-cid-5ivck5fi="true" data-astro-cid-syldxee3="true">  <svg width="12" height="12" viewBox="0 0 16 20" fill="none" xmlns="http://www.w3.org/2000/svg"> <path d="M6 0H12V2H16V20H0V2H4V0H6ZM12 4V6H4V4H2V18H14V4H12ZM10 4V2H6V4H10Z" fill="currentColor"></path> </svg>  </button>  <textarea class="[ hidden ]" id="copy-auth-token" readonly data-astro-cid-5ivck5fi="true" data-astro-cid-2vffnk23="true"></textarea>  </div> <small class="[ animate-fade-in-up ]" x-show="click_class === 'glow'" style="display: none" data-astro-cid-5ivck5fi>Copied!</small>  </div>   "
`;
